src/hotspot/share/code/codeBlob.hpp

```diff
@@ -749,16 +751,19 @@ class SafepointBlob: public SingletonBlob {
 
 class ProgrammableUpcallHandler;
 
-class OptimizedEntryBlob: public BufferBlob {
+class OptimizedEntryBlob: public RuntimeBlob {
```

Comment:
Some of it is discussed here: https://github.com/openjdk/panama-foreign/pull/617

Essentially, it is to avoid accidentally inheriting behavior from BufferBlob which we don't want. Also, BufferBlob currently expects a fixed-sized header (`sizeof(BufferBlob)`), while OptimizedEntryBlobs has fields, so we'd have to pass the real header size to the `BufferBlob` constructor, which is a bit messy.

It felt better to just cleanly break away from BufferBlob.